recipes/gromacs/2020/build.sh

```diff
@@ -3,26 +3,32 @@ mkdir build
 cd build
 
 ## See INSTALL of gromacs distro
-for ARCH in SSE2 AVX_256 AVX2_256 AVX_512; do \
-  cmake .. \
-  -DSHARED_LIBS_DEFAULT=OFF \
-  -DBUILD_SHARED_LIBS=OFF \
-  -DGMX_PREFER_STATIC_LIBS=YES \
-  -DGMX_BUILD_OWN_FFTW=OFF \
-  -DGMX_DEFAULT_SUFFIX=ON \
-  -DREGRESSIONTEST_DOWNLOAD=ON \
-  -DGMX_MPI=OFF \
-  -DGMX_GPU=ON \
-  -DGMX_USE_OPENCL=ON \
-  -DCMAKE_PREFIX_PATH=${PREFIX} \
-  -DGMX_INSTALL_PREFIX=${PREFIX} \
-  -DCMAKE_INSTALL_PREFIX=${PREFIX} \
-  -DGMX_SIMD=${ARCH} \
-  -DCMAKE_INSTALL_BINDIR=bin.${ARCH} \
-  -DCMAKE_INSTALL_LIBDIR=lib.${ARCH}
-  make -j${CPU_COUNT}
+for ARCH in SSE2 AVX_256 AVX2_256 AVX_512; do
+  cmake_args=(
+    -DSHARED_LIBS_DEFAULT=OFF
+    -DBUILD_SHARED_LIBS=OFF
+    -DGMX_PREFER_STATIC_LIBS=YES
```

Comment:
Why does this use static instead of shared libraries? The latter is more common/preferred in the Conda ecosystem.

Also, this package ships 4 `lib.*/gromacs.a` static libraries of \~ 36 MiB each, which are unused. If we expand the build matrix with different MPI variants, this means that we unnecessarily add > 400 MiB with each package build job.

Reply:
Thanks for all your help so far @mbargull.  I may have to go back to some of the developers who originally built this package as I just inherited the scripts form them and I am not sure why some of the build arguments are why they are.

I am not sure on the static vs shared so I will follow this up. 

The  lib.*/gromacs.a may have been there to allow other binaries to link against it, but this may not be needed and could be dropped in favour of .so.  - again I shall investigate this.

We build Gromacs optimised for 4 different SIMD types, it was decided building in this way was probably the best even though it is messy.

Reply:
Core gmx dev here... the author of that choice is unknown to us. It's certainly not needed for correctness, and other packaging systems don't need to build it that way.

A statically linked gmx binary is a trifle more performant, and someone who wanted that probably wants to install only the gmx executable and not libgromacs.a. However, I recommend bioconda build the shared library and install it alongside the gmx that links to it, so that also the Python API (called gmxapi) will build and install by default. Those strategies have comparable installation size.

Thus either

```suggestion
    -DSHARED_LIBS_DEFAULT=ON
    -DBUILD_SHARED_LIBS=ON
    -DGMX_PREFER_STATIC_LIBS=NO
```

or just drop those three lines to accept the defaults. Those happen to be the same for linux, and might be slightly more useful, e.g. if there's a platform where GROMACS doesn't support linking with shared libs (e.g. MSVC, I think).

Reply:
@mabraham, thanks for weighing in!